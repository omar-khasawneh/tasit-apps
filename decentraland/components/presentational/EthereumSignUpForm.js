import React from "react";
import { StyleSheet, View, TextInput } from "react-native";
import Colors from "@constants/Colors";

import {
  responsiveHeight,
  responsiveWidth,
  responsiveFontSize,
} from "react-native-responsive-dimensions";
import PropTypes from "prop-types";
import Button from "./Button";

export default class EthereumSignUpForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = { username: "" };
  }

  render() {
    const { onSignUp } = this.props;

    return (
      <React.Fragment>
        <View style={styles.userRow}>
          <View style={styles.userInputView}>
            <TextInput
              autoCorrect={false}
              autoCapitalize="none"
              style={styles.userInput}
              onChangeText={username => this.setState({ username })}
              value={this.state.username}
              placeholder="username"
              placeholderTextColor={Colors.placeholderTextColor}
              selectionColor={Colors.selection}
              keyboardAppearance="dark"
            />
          </View>
        </View>
        <View style={styles.buttonView}>
          <Button title="Continue" onPress={onSignUp} />
        </View>
      </React.Fragment>
    );
  }
}

EthereumSignUpForm.propTypes = {
  onSignUp: PropTypes.func.isRequired,
};

// TODO: Make the input box look a little nicer, maybe with a border

const styles = StyleSheet.create({
  userRow: { flexDirection: "row" },
  userInputView: { flex: 1, alignItems: "center" },
  userInput: {
    justifyContent: "flex-start",
    width: responsiveWidth(28),
    fontSize: responsiveFontSize(3),
    color: Colors.textColor,
    backgroundColor: Colors.formBackground,
  },
  buttonView: {
    flexDirection: "row",
    marginTop: responsiveHeight(5),
  },
});
