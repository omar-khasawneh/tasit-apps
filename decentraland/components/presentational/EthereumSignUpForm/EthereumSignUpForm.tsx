import React from "react";
import { StyleSheet, View, TextInput } from "react-native";
import Colors from "../../../constants/Colors";

import {
  responsiveHeight,
  responsiveWidth,
  responsiveFontSize,
} from "react-native-responsive-dimensions";

import Button from "../Button";

interface EthereumSignUpFormProps {
  onSignUp: any; // TODO: Change me to a function type
}

export default class EthereumSignUpForm extends React.Component<
  EthereumSignUpFormProps
> {
  constructor(props) {
    super(props);
    this.state = { username: "" };
  }

  onUsernameChange = username => this.setState({ username });

  render() {
    const { onSignUp } = this.props;

    return (
      <React.Fragment>
        <View style={styles.userRow}>
          <View style={styles.userInputView}>
            <UsernameTextInput
              onChange={this.onUsernameChange}
              username={this.state.username}
            />
          </View>
        </View>
        <View style={styles.buttonView}>
          <Button title="Continue" onPress={onSignUp} />
        </View>
      </React.Fragment>
    );
  }
}

const styles = StyleSheet.create({
  buttonView: {
    flexDirection: "row",
    marginTop: responsiveHeight(5),
  },
  userInput: {
    backgroundColor: Colors.formBackground,
    color: Colors.textColor,
    fontSize: responsiveFontSize(3),
    justifyContent: "flex-start",
    width: responsiveWidth(48),
  },
  userInputView: { alignItems: "center", flex: 1 },
  userRow: { flexDirection: "row" },
});

interface UsernameTextInputProps {
  onChange: any; // TODO: Change me to a function type
  username: string;
}

export const UsernameTextInput: React.SFC<UsernameTextInputProps> = ({
  onChange,
  username,
}) => {
  return (
    <TextInput
      autoCorrect={false}
      autoFocus={true}
      autoCapitalize="none"
      style={styles.userInput}
      onChangeText={onChange}
      value={username}
      placeholder="username"
      placeholderTextColor={Colors.placeholderTextColor}
      selectionColor={Colors.selection}
      keyboardAppearance="dark"
    />
  );
};
