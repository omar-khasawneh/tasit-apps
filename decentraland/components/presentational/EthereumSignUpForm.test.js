import React from "react";
import NavigationTestUtils from "react-navigation/NavigationTestUtils";
import { shallow } from "enzyme";
import EthereumSignUpForm from "./EthereumSignUpForm";

describe("EthereumSignUpForm", () => {
  jest.useFakeTimers();
  let wrapper;
  let onSignUp;

  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
    onSignUp = jest.fn();
    wrapper = shallow(<EthereumSignUpForm onSignUp={onSignUp} />);
  });

  it("renders the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("creates a wallet - calling function", () => {
    wrapper
      .find("Button")
      .find({ title: "Continue" })
      .simulate("press");

    expect(onSignUp.mock.calls.length).toBe(1);
  });
});
