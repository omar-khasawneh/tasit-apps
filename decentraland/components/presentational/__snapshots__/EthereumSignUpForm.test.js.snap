// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EthereumSignUpForm renders the component 1`] = `
<Fragment>
  <Component
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="username"
        style={
          Object {
            "fontSize": 45.89389937671454,
            "justifyContent": "flex-start",
            "width": 210.00000000000003,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </Component>
  </Component>
  <Component
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 66.7,
      }
    }
  >
    <Button
      onPress={[MockFunction]}
      title="Continue"
    />
  </Component>
</Fragment>
`;
